test/parallel/test-internal-process-execution.js

```diff
@@ -0,0 +1,37 @@
+// Flags: --expose-internals --experimental-modules
+'use strict';
+
+const { mustCall } = require('../common');
+const { evalModule } = require('internal/process/execution');
+const globalConsole = require('internal/console/global');
+const { strictEqual } = require('assert');
+
+function wrapConsole(fnName, action) {
+  const originalFn = globalConsole[fnName];
+  globalConsole[fnName] = (input) => {
+    action(input);
+    originalFn.call(globalConsole, input);
+    globalConsole[fnName] = originalFn;
+  };
+}
+
+const test = (code, shouldFail, type, assert) => {
+  wrapConsole(type, mustCall(assert, 1));
+  if (shouldFail) {
+    const originalExit = process.exit;
+    process.exit = mustCall((code) => {
+      strictEqual(code, 1);
+      process.exit = originalExit;
+    }, 1);
+  }
+  evalModule(code, true);
+};
+
+// BC! Unpredictable order of promise execution
+// due to hidden promise inside eval
+test('"use strict";', false, 'log',
```

Comment:
I have here unpredictable execution order, because `promise` is hidden inside `evalModule` function.
The solution I see is to `return` promise from `evalModule` then all tests could be chained